### PR TITLE
infoschema: checks if DBInfo read from Meta is nil.

### PR DESCRIPTION
**Pull request by coocood**
Fix panic.

### CODE REVIEW

**Comment by coocood** [comment]
@zimulala @hanfei1991 PTAL

**Comment by hanfei1991** [comment]
LGTM

File: infoschema/infoschema_test.go [hunk 1]
```diff
@@ -171,6 +174,14 @@ func (*testSuite) TestT(c *C) {
 	c.Assert(tb, NotNil)
 }
 
+func checkApplyCreateNonExistsSchemaDoesNotPanic(c *C, store kv.Storage, builder *infoschema.Builder) {
+	txn, err := store.Begin()
+	c.Assert(err, IsNil)
+	m := meta.NewMeta(txn)
+	err = builder.ApplyDiff(m, &model.SchemaDiff{Type: model.ActionCreateSchema, SchemaID: 999})
+	c.Assert(infoschema.ErrDatabaseNotExists.Equal(err), IsTrue)
```
Comment (zimulala) [inline infoschema/infoschema_test.go]: This test is only for ErrDatabaseNotExists， add test for  ErrTableNotExists.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**